frb/halos/hmf.py

```diff
@@ -0,0 +1,389 @@
+import warnings
+
+import numpy as np
+from scipy.interpolate import InterpolatedUnivariateSpline as IUS
+
+from astropy import units
+from astropy.cosmology import Planck15 as cosmo
+from astropy import constants
+from astropy.cosmology import z_at_value
+from astropy.table import Table
+
+from frb.halos.models import ModifiedNFW, ICM
+
+from IPython import embed
+
+def init_hmf():
+    """
+    Initialize the Aemulus Halo Mass Function
+
+    WARNING: This uses the original version which codes Tinker+2008
+    We may refactor to use the more accurate, new version
+
+    Returns:
+
+    """
+    # Hidden here to avoid it becoming a dependency
+    import hmf_emulator
+    # Setup HMF
+    # https://github.com/astropy/astropy/blob/master/astropy/cosmology/parameters.py
+    #sigma8 = 0.8159
+    ns = 0.9667
+    Neff = 3.046
+    #cosmo_dict = {"om":cosmo.Om0,"ob":cosmo.Ob0,"ol":1.-cosmo.Om0,"ok":0.0,
+    #              "h":cosmo.h,"s8":sigma8,"ns":ns,"w0":-1.0,"Neff":Neff} # "wa":0.0 is assumed internally
+    cosmo_dict = {"omega_cdm":(cosmo.Om0-cosmo.Ob0)*cosmo.h**2,
+                  "omega_b":cosmo.Ob0*cosmo.h**2,"ok":0.0,
+                  "ln10As": 3.098, # THIS REPLACES sigma8
+                  "H0":cosmo.H0.to('km/(s*Mpc)').value,
+                  "n_s":ns,"w0":-1.0,"N_eff":Neff} # "wa":0.0 is assumed internally
+    hmfe = hmf_emulator.hmf_emulator()
+    hmfe.set_cosmology(cosmo_dict)
+    # Return
+    return hmfe
+
+# Storing for use
+#  Needs to come after def init_hmf()
+try:
+    import hmf_emulator
+except:
```

Comment:
The exception should warn the user about an unsuccessful import. Otherwise, down the line, I'm afraid it'll give you an inexplicable "hmfe not defined" error.

Reply:
warning raised